### ISSUE TITLE
Configure on a namespace basis the fields to index in the thing-search

### ISSUE THREAD

**Issue by thjaeckle** [eclipse-ditto/ditto#1521]
Currently, the complete JSON of a thing (including all attributes+features) is automatically indexed in the search.
We learned from input of the community that this might not always be wished as this could lead to a lot of load on the search DB - which could e.g. be prevented if some fields are never used in the search "fields".

**Comment by an1310** [comment]
This is of strategic interest to us -- when would be a good time to go over requirements/design ideas for a possible contribution?

**Comment by thjaeckle** [comment]
@an1310 I will try to find out where this could be placed the best.
As far as I remember there was an extension point for just that purpose used by Bosch's commercial offering of Ditto.

**Comment by thjaeckle** [comment]
@an1310 I took myself a closer look and I think I found the mentioned extension point to do index only custom fields in namespaces.

I think to remember that we explicitly added `CachingSignalEnrichmentFacade` extension point in Ditto to retrieve only selectively parts of "things" (based on their namespace "search index configuration") when processing - retrieving a "partial thing" in order to update the search index:
https://github.com/eclipse-ditto/ditto/blob/022dffee0c1a21cedd57b74310683db5d7ce881e/internal/models/signalenrichment/src/main/java/org/eclipse/ditto/internal/models/signalenrichment/DittoCachingSignalEnrichmentFacade.java#L113-L158

We would need a custom `JsonFieldSelector` configured per namespace name, defining which fields of a "thing" to retrieve (and cache) for updating the search index.
The configuration could look like:
* a list/map of namespaces
   * namespace
      * indexedFields as `JsonFieldSelector`

Example what I have in mind (to put in the `search.conf` as configuration of the mentioned extension point):
https://github.com/eclipse-ditto/ditto/blob/022dffee0c1a21cedd57b74310683db5d7ce881e/thingsearch/service/src/main/resources/search.conf#L11

```
namespaces = [
  {
    namespace-name = "org.eclipse.ditto.foo"
    indexed-fields = [ 
      "thingId", 
      "policyId", 
      "_revision", 
      "_created", 
      "_modified", 
      "attributes/indexed-one"
      "attributes/complex-jsonobj"
    ]
  }
]
```

I hope that helps a little. Maybe a Ditto committer still at Bosch can share more details how this can be done, based on the identified extension point.
